mypy/checkexpr.py

```diff
@@ -3724,6 +3724,10 @@ def visit_assert_type_expr(self, expr: AssertTypeExpr) -> Type:
         )
         target_type = expr.type
         if not is_same_type(source_type, target_type):
+            if not self.chk.in_checked_function():
+                self.msg.note(
+                    "'assert_type' always outputs 'Any' in unchecked functions", expr.expr
```

Comment:
I don't think this is quite right; assert_type() doesn't output anything.

Reply:
Ah, you mean in terms of wording? Would "'assert_type(x, y) always evaluate `x` as `Any`" ?

Reply:
Oh, I see now.. if doesn't actually output a message in the test.

When I actually run it from my branch I saw

```console
$ mypy func_scope.py 
func_scope.py:5: error: Expression is of type "Any", not "Literal[42]"
func_scope.py:5: note: 'assert_type' always outputs 'Any' in unchecked functions
Found 1 error in 1 file (checked 1 source file)
```

so it looked like my fix worked

Reply:
Yes, your code looks correct in the sense that the note gets output, but I don't think the wording is good. I like @hauntsaninja's suggestion below.

Reply:
Gotchya!

I am not sure why my test is failing. It looks like it has the same form as the other test cases, but no message is emitted: https://github.com/python/mypy/runs/8241823231?check_suite_focus=true#step:8:295